### PR TITLE
Recvfd Registry should expect Endpoint name on return

### PR DESCRIPTION
**Pull request by haiodo**
Recvfd Registry should expect Endpoint name on return

Signed-off-by: Andrey Sobolev <haiodo@gmail.com>

### CODE REVIEW

File: pkg/registry/common/recvfd/server.go [hunk 1]
```diff
@@ -73,6 +78,10 @@ func (r *recvfdNseServer) Register(ctx context.Context, endpoint *registry.Netwo
 	}
 	returnedEndpoint = returnedEndpoint.Clone()
 
+	if endpointName != returnedEndpoint.Name {
```
Comment (edwarnicke) [inline pkg/registry/common/recvfd/server.go:81]: Curious, when would endpointName != returnedEndpoint.Name?

Comment (denis-tingajkin) [reply pkg/registry/common/recvfd/server.go:81]: I feel because remote side could set unique identity for incoming endpoint

Comment (edwarnicke) [reply pkg/registry/common/recvfd/server.go:81]: Got it.  I was curious if there was a specific use case in mind, or just an abundance of caution :)

Comment (haiodo) [reply pkg/registry/common/recvfd/server.go:81]: If they are same we are during heal Register, to update we alive. And if not we expect remote side to update with a proper name.

**Pull request merged**